pkg/deploy/deploy_test.go

```diff
@@ -1298,3 +1299,56 @@ func TestDeployConfigGraph_CollectsAllErrors(t *testing.T) {
 	})
 
 }
+
+func TestDeployConfigFF(t *testing.T) {
+	dummyClientSet := client.DummyClientSet
+	c := dynatrace.EnvironmentClients{
+		dynatrace.EnvironmentInfo{Name: "env"}: &dummyClientSet,
+	}
+	tests := []struct {
+		name              string
+		projects          []project.Project
+		featureFlag       string
+		configType        config.TypeID
+		expectedErrString string
+	}{
+		{
+			name: "segments FF test",
+			projects: []project.Project{
+				{
+					Configs: project.ConfigsPerTypePerEnvironments{
+						"env": project.ConfigsPerType{
+							"p1": {
+								config.Config{
+									Type:        config.Segment{},
+									Environment: "env",
+									Coordinate: coordinate.Coordinate{
+										Project:  "p1",
+										Type:     "type",
+										ConfigId: "config1",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			featureFlag: featureflags.Segments.EnvName(),
+			configType:  config.SegmentID,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name+" | FF Enabled", func(t *testing.T) {
+			t.Setenv(tt.featureFlag, "true")
+			err := deploy.Deploy(context.TODO(), tt.projects, c, deploy.DeployConfigsOptions{})
```

Comment:
minor nit, because I don't find too much: `context.Background()` would be the correct one to use ;)
https://pkg.go.dev/context#Background

However, starting with go1.24, it will be anyway `testing.Context`, so no need to worry here :)